SMLHelper/Options/ConfigModOptions.cs

```diff
@@ -0,0 +1,952 @@
+            {
+                ProcessFieldOrProperty(field, MemberType.Field, field.FieldType);
+            }
+
+            foreach (MethodInfo method in typeof(T).GetMethods(bindingFlags | BindingFlags.Static)
+                .Where(MemberIsDeclaredInConfigFileSubclass) // Only care about members declared in a subclass of ConfigFile
+                .Where(MemberIsNotIgnored)) // Filter out explicitly ignored members
+            {
+                ProcessMethod(method);
+            }
+
+            Logger.Debug($"[ConfigModOptions] Found {modOptionsMetadata.Count()} options to add to the menu.");
+        }
+
+        /// <summary>
+        /// Gets the <see cref="MenuAttribute"/> defined on the <typeparamref name="T"/>, or
+        /// a default <see cref="MenuAttribute"/> with its name automatically parsed from the Display Name
+        /// of the QMod that <typeparamref name="T"/> is defined in.
+        /// </summary>
+        /// <returns></returns>
+        private MenuAttribute GetMenuAttributeOrDefault()
+        {
+            if (Attribute.IsDefined(typeof(T), typeof(MenuAttribute), true))
+                return typeof(T).GetCustomAttributes(typeof(MenuAttribute), true).SingleOrDefault() as MenuAttribute;
+            else
+                return new MenuAttribute(QMod.DisplayName);
+        }
+
+        /// <summary>
+        /// Checks whether a given <see cref="MemberInfo"/> is declared in any subclass of <see cref="ConfigFile"/>.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> to check.</param>
+        /// <returns>Whether the given <see cref="MemberInfo"/> is declared in any subclass of <see cref="ConfigFile"/>.</returns>
+        private static bool MemberIsDeclaredInConfigFileSubclass(MemberInfo memberInfo)
+            => memberInfo.DeclaringType.IsSubclassOf(typeof(ConfigFile));
+
+        /// <summary>
+        /// Checks whether a given <see cref="MemberInfo"/> should be ignored when generating the options menu, based on whether
+        /// the member has a declared <see cref="IgnoreMemberAttribute"/>, or the <see cref="menuAttribute"/>'s
+        /// <see cref="MenuAttribute.IgnoreUnattributedMembers"/> property.
+        /// </summary>
+        /// <param name="memberInfo">The <see cref="MemberInfo"/> to check.</param>
+        /// <returns>Whether the given <see cref="MemberInfo"/> member should be ignored when generating the options menu.</returns>
+        private bool MemberIsNotIgnored(MemberInfo memberInfo)
+        {
+            if (Attribute.IsDefined(memberInfo, typeof(IgnoreMemberAttribute)))
+            {
+                return false;
+            }
+            else if (!menuAttribute.IgnoreUnattributedMembers)
+            {
+                if (memberInfo is MethodInfo)
+                {
+                    if (Attribute.IsDefined(memberInfo, typeof(ButtonAttribute), true))
```

Comment:
Specifying it to be explicit, so that everyone knows without having to go and look it up.

Reply:
Well, I don't think that everyone even know what that parameter is for. I'm using that `Attribute.IsDefined` in my code and I needed to "go and look it up" to find out why I don't use that parameter and should I use it :)

Reply:
![image](https://user-images.githubusercontent.com/45315526/89334533-91eab680-d68e-11ea-9ca3-55a635f2bbe5.png)

I'm talking about people working on the actual codebase. I'd rather be explicit than not. Just because it is the default does not mean you **should not define it,** it just means it is the default if you don't. So I'm not sure why you thought it necessary to leave a review telling me that it is default true as if you think I should remove it?

Reply:
It have overload without that param and that overload uses `inherit = true`. It's just harder to read, but if you think that that `true` absolutely needed to be there, ok then.